st2common/st2common/logging/formatters.py

```diff
@@ -0,0 +1,209 @@
+
+        return result
+
+
+class ConsoleLogFormatter(BaseExtraLogFormatter):
+    """
+    Formatter which attaches all the attributes from the "extra" dictionary as key=value pairs to
+    the end of the log message.
+
+    For example:
+
+        LOG.info('Hello world', extra={'_id': 1, '_path': '/fooo'})
+
+    Result:
+
+        Hello World (id=1,path=foo)
+    """
+
+    def format(self, record):
+        attributes = self._get_extra_attributes(record=record)
+        attributes = self._format_extra_attributes(attributes=attributes)
+        attributes = self._dict_to_str(attributes=attributes)
+
+        # Call the parent format method so the final message is formed based on the "format"
+        # attribute in the config
+        msg = super(ConsoleLogFormatter, self).format(record)
+
+        if attributes:
+            msg = '%s (%s)' % (msg, attributes)
+
+        return msg
+
+    def _dict_to_str(self, attributes):
+        result = []
+        for key, value in six.iteritems(attributes):
+            item = '%s=%s' % (key[1:], repr(value))
+            result.append(item)
+
+        result = ','.join(result)
+        return result
+
+
+class GelfLogFormatter(BaseExtraLogFormatter):
+    """
+    Formatter which formats messages as GELF 2 - https://www.graylog.org/resources/gelf-2/
+
+    For example:
+
+        LOG.info('Hello world', extra={'_id': 1, '_path': '/fooo'})
+
+    Result:
+
+        {
+            "version": "1.1",
+            "level": 6,
+            "timestamp": 1426590583,
+            "_python": {
+                "process": 11277,
+                "module": "__init__",
+                "funcName": "db_setup",
+                "processName": "MainProcess",
+                "lineno": 28,
+                "filename": "__init__.py"
+            },
+            "host": "vagrant-ubuntu-trusty-64",
+            "full_message": "2015-03-17 11:09:43,507 INFO [-] Hello world",
+            "_path": "/fooo",
+            "_id": 1,
+            "short_message": "Hello world"
+        }
+    """
+
+    # Maps python log level to syslog / gelf log level
+    PYTHON_TO_GELF_LEVEL_MAP = {
+        50: 2,  # critical -> critical
+        40: 3,  # error -> error
+        30: 4,  # warning -> warning
+        20: 6,  # info -> informational
+        10: 7,  # debug -> debug
+        0: 6,  # notset -> information
+    }
+    DEFAULT_LOG_LEVEL = 6  # info
+
+    def format(self, record):
+        attributes = self._get_extra_attributes(record=record)
+        attributes = self._format_extra_attributes(attributes=attributes)
+
+        msg = record.msg
+        exc_info = record.exc_info
+        now = int(time.time())
+        level = self.PYTHON_TO_GELF_LEVEL_MAP.get(record.levelno, self.DEFAULT_LOG_LEVEL)
+
+        common_attributes = self._get_common_extra_attributes(record=record)
+        full_msg = super(GelfLogFormatter, self).format(record)
+
+        data = {
+            'version': '1.1',
```

Comment:
I was thinking of declaring a module level constant with a base dictionary template, but then we would really need to use copy.copy inside the format calls which is less than ideal.